duckduckgo/Android
# Migrate from @OnLifecycleEvent: app module

Task/Issue URL: https://app.asana.com/0/0/1202260594569397/f

### Description

@OnLifecycleEvent is now deprecated. Updated code in app module to use `DefaultLifecycleObserver`

### Steps to test this PR

- Install from this branch
- Launch the app
- Check in the Logcat that the following Logs appear:

- [ ] Registering application lifecycle observer: com.duckduckgo.app.FileDownloadBroadcastReceiver
- [ ] Registering application lifecycle observer: com.duckduckgo.app.RealWebViewHttpAuthStore
- [ ] Registering application lifecycle observer: com.duckduckgo.app.ServiceWorkerLifecycleObserver
- [ ] Registering application lifecycle observer: com.duckduckgo.app.ShortcutReceiver
- [ ] Registering application lifecycle observer: com.duckduckgo.app.TrackerDataDevReceiverRegister
- [ ] Registering application lifecycle observer: com.duckduckgo.app.DataClearerForegroundAppRestartPixel
- [ ] Registering application lifecycle observer: com.duckduckgo.app.LottieFireAnimationLoader
- [ ] Registering application lifecycle observer: com.duckduckgo.app.FlipperInitializer
- [ ] Registering application lifecycle observer: com.duckduckgo.app.AppDataLoader
- [ ] Registering application lifecycle observer: com.duckduckgo.app.AppInstallSharedPreferences
- [ ] Registering application lifecycle observer: com.duckduckgo.app.MigrationLifecycleObserver
- [ ] Registering application lifecycle observer: com.duckduckgo.app.AppEnjoymentAppCreationObserver
- [ ] Registering application lifecycle observer: com.duckduckgo.app.AppShortcutCreatorLifecycleObserver
- [ ] Registering application lifecycle observer: com.duckduckgo.app.HttpsUpgraderImpl
- [ ] Registering application lifecycle observer: com.duckduckgo.app.AndroidWorkScheduler
- [ ] Registering application lifecycle observer: com.duckduckgo.app.AppConfigurationSyncer
- [ ] Registering application lifecycle observer: com.duckduckgo.app.NotificationRegistrar
- [ ] Registering application lifecycle observer: com.duckduckgo.app.AppUserStageStore
- [ ] Registering application lifecycle observer: com.duckduckgo.app.EnqueuedPixelWorker
- [ ] Registering application lifecycle observer: com.duckduckgo.app.TrackersDbCleanerScheduler
- [ ] Registering application lifecycle observer: com.duckduckgo.app.AtbInitializer
- [ ] Registering application lifecycle observer: com.duckduckgo.app.OfflinePixelScheduler
- [ ] Registering application lifecycle observer: com.duckduckgo.app.RxPixelSender
- [ ] Registering application lifecycle observer: com.duckduckgo.app.ResourceSurrogateLoader
- [ ] Registering application lifecycle observer: com.duckduckgo.app.TabsDbSanitizer
- [ ] Registering application lifecycle observer: com.duckduckgo.app.AppStartUpTracer
- [ ] Registering application lifecycle observer: com.duckduckgo.app.TrackerDataLoader
- [ ] Registering application lifecycle observer: com.duckduckgo.app.AppDaysUsedRecorder
- [ ] Registering application lifecycle observer: com.duckduckgo.app.AppEmailWaitlistCodeFetcher
- [ ] Registering application lifecycle observer: com.duckduckgo.app.AppTrackingProtectionWaitlistCodeFetcher
- [ ] Registering application lifecycle observer: com.duckduckgo.app.FavoritesObserver
- [ ] Registering application lifecycle observer: com.duckduckgo.app.WidgetAddedReceiver
